ladybug/windrose.py

```diff
@@ -0,0 +1,953 @@
+        * inner_boundary_circle
+        * frequency_spacing_distance
+        * frequency_maximum
+        * label_point_size_factor
+        * label_tick_size_factor
+        * orientation_labels
+        """
+
+    def __init__(self, direction_data_collection, analysis_data_collection,
+                 number_of_directions=8):
+
+        """Initialize windrose plot."""
+        # Check the input objects
+        acceptable_colls = (HourlyContinuousCollection, HourlyDiscontinuousCollection)
+        assert isinstance(direction_data_collection, acceptable_colls), 'Windrose ' \
+            'direction_data_collection must be a HourlyContinuousCollection or ' \
+            'HourlyDiscontinuousCollection. Got {}.'.format(
+                type(direction_data_collection))
+        assert isinstance(analysis_data_collection, acceptable_colls), 'Windrose ' \
+            'analysis_data_collection must be a HourlyContinuousCollection or ' \
+            'HourlyDiscontinuousCollection. Got {}.'.format(
+                type(analysis_data_collection))
+
+        # Ensure the analysis period of the data collection has been validated
+        if not direction_data_collection.validated_a_period:
+            direction_data_collection = \
+                direction_data_collection.validate_analysis_period()
+        if not direction_data_collection.validated_a_period:
+            analysis_data_collection = \
+                analysis_data_collection.validate_analysis_period()
+
+        # Assign the inputs as read-only properties of this data collection
+        self._direction_data_collection = direction_data_collection.to_immutable()
+        self._analysis_data_collection = analysis_data_collection.to_immutable()
+        self._number_of_directions = number_of_directions
+
+        # Compute the windrose data and associated read-only properties
+        self._angles = linspace(0, 360, number_of_directions + 1)
+        self._data, self._zero_count = \
+            WindRose._compute_windrose_data(self.direction_values, self.analysis_values,
+                                            number_of_directions, self.angles,
+                                            (0, 360))
+
+        # Editable public properties for visualization
+        self._legend = None
+        self._legend_parameters = None
+        self._frequency_spacing_distance = None
+        self._frequency_maximum = None
+        self._label_point_size_factor = None
+        self._label_tick_size_factor = None
+        self._show_stack = None
+        self._show_zeros = None
+        self._base_point = None
+
+        # Fixed properties for visualization that only need to be computed once
+        self._bin_vectors = None
+        self._zeros_per_bin = None
+        self._orientation_labels = None
+        self._real_freq_max = None
+
+    @property
+    def base_point(self):
+        """An optional Point3D used as the bottom left corner of the windrose geometry.
```

Comment:
@chriswmackey I originally tried this but the GraphicContainer class expects a Point3D for min and max_point. Should we change the GraphicContainer class or keep as is?

Reply:
Ref: Line 47 - 50 in graphic.py

> /app/ladybug/ladybug/graphic.py in __init__(self, values, min_point, max_point, legend_parameters, data_type, unit)
     46         # check the inputs
     47         assert isinstance(min_point, Point3D), \
---> 48             'min_point should be a ladybug Point3D. Got {}'.format(type(min_point))
     49         assert isinstance(max_point, Point3D), \
     50             'max_point should be a ladybug Point3D. Got {}'.format(type(max_point))

Reply:
The `GraphicContainer` is written that way because it has to deal with a wide variety of visualizations. For the wind rose, I think we should probably keep it as a Point2D publicly and, under the hood, we convert it to a Point3D with 0 for the z value.